packages/jwt/src/types.ts

```diff
@@ -0,0 +1,80 @@
+export type Scope = "send-manifest" | "get-manifest" | "logs" | "shell" | "events" | "status" | "restart" | "hostname-migrate" | "ip-migrate";
+export type Access = "full" | "granular" | "scoped";
+
+export interface JwtTokenOptions {
+  /** Akash address of the lease(s) owner, e.g., akash1abcd... (44 characters) */
+  iss: string;
+  /** Token issuance timestamp as Unix time (seconds since 1970-01-01T00:00:00Z). Should be <= exp and >= nbf. */
+  iat: number;
+  /** Not valid before timestamp as Unix time (seconds since 1970-01-01T00:00:00Z). Should be <= iat. */
+  nbf: number;
+  /** Expiration timestamp as Unix time (seconds since 1970-01-01T00:00:00Z). Should be >= iat. */
+  exp: number;
+  /** Unique identifier for the JWT, used to prevent token reuse. */
+  jti?: string;
+  /** Version of the JWT specification (currently fixed at v1). */
+  version: "v1";
+  /** Access control configuration for leases */
+  leases: {
+    /** Access level for the lease: 'full' for unrestricted access to all actions, 'granular' for provider-specific permissions. */
+    access: Access;
+    /** Global list of permitted actions across all owned leases (no duplicates). Applies when access is 'full'. */
+    scope?: Scope[];
+    /** Required if leases.access is 'granular'; defines provider-specific permissions. */
```

Comment:
suggestion(non-blocking): it's possible to define it with tagged union:

```ts
type Leases = FullAccessToLeases | GranularAccessToLeases

interface FullAccessToLeases {
  access: 'full';
  scope: Scope[];
}

interface GranularAccessToLeases {
  access: 'granular';
  permissions: ...
}
```

This will validate the correct object shape at build time with help of TypeScript